### PR TITLE
feat(tasking): Implement notification hooks for ImagingRequest and LabRequest updates

### PR DESCRIPTION
**Pull request by khoi-bes**
### Changes

- Move pushing notification logic to afterUpdate hooks

### Deploys

- [ ] **Deploy to Tamanu Internal** 

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

### CODE REVIEW

**Review by namhoangbes — APPROVED** [review]

File: packages/shared/src/models/Notification.js [hunk 1]
```diff
@@ -89,16 +89,17 @@ export class Notification extends Model {
...
           const encounter = await models.Encounter.findByPk(metadata.encounterId);
-          patientId = encounter.patientId;
+          patientId = encounter?.patientId;
           break;
         }
         case NOTIFICATION_TYPES.LAB_REQUEST: {
           userId = metadata.requestedById;
           const encounter = await models.Encounter.findByPk(metadata.encounterId);
-          patientId = encounter.patientId;
+          patientId = encounter?.patientId;
```
Comment (edmofro) [inline packages/shared/src/models/Notification.js:98]: Why is the `?` necessary? There should never be a time where an imaging request or lab request _doesn't_ have an encounter, right?

**Review by edmofro — COMMENTED** [review]
I have another question on this one too

File: packages/shared/src/models/Notification.js (hunk 1 above)
Comment (khoi-bes) [reply packages/shared/src/models/Notification.js:98]: yes your are right, but in theory, the `patientId` can be null in the encounters table. As I remember, it was failed in some test cases so I have to use the `?`

Comment (edmofro) [reply packages/shared/src/models/Notification.js:98]: Oh right it's nullable in the DB! I see that now. It's bad that we have that as nullable, have added a maintenance sprint item [here](https://beyond-essential.slab.com/posts/tamanu-maintenance-sprint-items-j1ac0rkb#hapt2-new) to fix that.

However, that should never be the case in reality though - all encounters should have a `patient_id`, and I've never seen one without. If test cases were failing that means the tests need fixing, not the code. We should _never_ add defensive code unless there is a real use case where the data should be null/undefined. This isn't one of those cases.

Can you please update it to remove the defensive programming and fix any tests that break as a result?

Comment (khoi-bes) [reply packages/shared/src/models/Notification.js:98]: alright, let me fix

Comment (edmofro) [reply packages/shared/src/models/Notification.js:98]: We shouldn't use a js logical change to fix a type error

**Pull request merged**